### PR TITLE
Fixed an issue where environment constants would prevent optional consta...

### PR DESCRIPTION
**Pull request by danwkennedy**
...nts from getting written to the output file.

### CODE REVIEW

File: gulp-ng-config.js [hunk 1]
```diff
@@ -40,13 +40,13 @@ function gulpNgConfig (moduleName, configuration) {
...
     }
 
-    jsonObj = _.merge({}, jsonObj, configuration.constants || {});
-
     // select the environment in the configuration
     if (configuration.environment && jsonObj.hasOwnProperty(configuration.environment)) {
       jsonObj = jsonObj[configuration.environment];
     }
 
+    jsonObj = _.merge({}, jsonObj, configuration.constants || {});
```
Comment (ajwhite) [inline gulp-ng-config.js]: Oh man, yeah good call :+1:

**Comment by ajwhite** [comment]
Ty for fixing this oversight!

Closes https://github.com/ajwhite/gulp-ng-config/issues/21

**Comment by clabnet** [comment]
I'm not sure that working:
on win 10, gulp-ng-config ver. 1.2.1, with this .json:

{
  "dev": {
    "EnvironmentConfig": {
      "api": "http://localhost/"
    }
  },
  "production": {
    "EnvironmentConfig": {
      "api": "https://api.production.com/"
    }
  }
}

the Gulp task :


gulp.task('ngConfig', function () {
  return gulp.src(config.configJson)
    .pipe(gulpNgConfig('dcc.config'), {
      environment: 'production'
    })
    .pipe(gulp.dest(config.scripts))
});

my expected result would be:

angular.module('dcc.config', [])
.constant('production', {"EnvironmentConfig":{"api":"https://api.production.com/"}});

and instead it is :

angular.module('dcc.config', [])
.constant('dev', {"EnvironmentConfig":{"api":"http://localhost/"}})
.constant('production', {"EnvironmentConfig":{"api":"https://api.production.com/"}});

(both entries)


Maybe I did not understand how to use it?

Thank you, bye

**Comment by ajwhite** [comment]
It looks like your gulpfile has an incorrect syntax:

```js
.pipe(gulpNgConfig('dcc.config'), {
  environment: 'production'
})
```

You're passing the configuration as a second argument to `pipe`, and not `gulpNgConfig`

it should be
```js
.pipe(gulpNgConfig('dcc.config', {
  environment: 'production'
}))
```

**Comment by clabnet** [comment]
You are right, it is my mistake. It run, grazie.

**Comment by ajwhite** [comment]
:smile: :+1:

**Pull request merged**